docs/upgrading.md

```diff
@@ -0,0 +1,63 @@
+This page provides some tipps of how to upgrade between versions.
+
+## v0.3.3 to v0.4.0
+
+- Message sending in the skills has been updated. In the past you had to construct messages, then serialize them and place them in an envelope:
+
+``` python
+cfp_msg = FipaMessage(...)
+self.context.outbox.put_message(
+    to=opponent_addr,
+    sender=self.context.agent_address,
+    protocol_id=FipaMessage.protocol_id,
+    message=FipaSerializer().encode(cfp_msg),
+)
+# or
+cfp_msg = FipaMessage(...)
+envelope = Envelope(
+    to=opponent_addr,
+    sender=self.context.agent_address,
+    protocol_id=FipaMessage.protocol_id,
+    message=FipaSerializer().encode(cfp_msg),
+)
+self.context.outbox.put(envelope)
+```
+
+Now this has been simplified to:
+``` python
+cfp_msg = FipaMessage(...)
+cfp_msg.counterparty = opponent_addr
```

Comment:
question: what happens if you forget to define the counterparty?

Reply:
It does not get sent (error). We are going to simplify this further when we go back to the dialogue models.